src/Stryker.Core/Stryker.Core/Baseline/AzureBaselineProvider.cs

```diff
@@ -0,0 +1,179 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Stryker.Core.Logging;
+using Stryker.Core.Options;
+using Stryker.Core.Reporters.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Stryker.Core.Baseline
+{
+    public class AzureBaselineProvider : IBaselineProvider
+    {
+        private readonly StrykerOptions _options;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger<AzureBaselineProvider> _logger;
+        public AzureBaselineProvider(StrykerOptions options, HttpClient httpClient = null)
+        {
+            _options = options;
+            _httpClient = httpClient ?? new HttpClient();
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<AzureBaselineProvider>();
+        }
+
+        public async Task<JsonReport> Load(string version)
+        {
+            var subdir = $"strykerOutput/baselines/{version}";
+            var filename = "stryker-report.json";
+            var url = new Uri($"{_options.AzureFileStorageUrl}/{subdir}/{filename}?sv={_options.AzureSAS}");
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, url);
+
+            requestMessage.Headers.Add("x-ms-date", "now");
+
+            var response = await _httpClient.SendAsync(requestMessage);
+
+            if (response.StatusCode == HttpStatusCode.OK)
+            {
+                var content = await response.Content.ReadAsStringAsync();
+
+                return JsonConvert.DeserializeObject<JsonReport>(content);
+            }
+            return null;
+        }
+
+        public async Task Save(JsonReport report, string version)
+        {
+            var existingReport = await Load(version);
+
+            var reportJson = report.ToJson();
+
+            int byteSize;
+            if (existingReport == null)
+            {
+                byteSize = Encoding.UTF8.GetByteCount(report.ToJson());
+
+                var succesfullyCreatedDirectory = await CreateDirectory(version);
+
+                if (!succesfullyCreatedDirectory)
+                {
+                    return;
+                }
+
+                var successfullyAllocated = await AllocateFileLocation(byteSize, version);
+
+                if (!successfullyAllocated)
+                {
+                    return;
+                }
+            }
+            else
+            {
+                byteSize = Encoding.UTF8.GetByteCount(existingReport.ToJson());
```

Comment:
If a report already exists, you set the byteSize for the upload to the size of the old report. What happens if the new report is larger? Do the extra bytes just get chopped off resulting in a malformed report?

Reply:
This is now fixed by doing an extra http call to azure